src/array/cpu/spmat_op_impl_coo.cc

```diff
@@ -312,26 +313,65 @@ CSRMatrix COOToCSR(COOMatrix coo) {
   NDArray ret_indices;
   NDArray ret_data;
 
-  bool row_sorted = coo.row_sorted;
-  bool col_sorted = coo.col_sorted;
+  const bool row_sorted = coo.row_sorted;
```

Comment:
Should we also change `cuda/spmat_op_impl_coo.cu`?

Reply:
The implementation in `cuda/coo2csr.cu` only operates on the sorted COO (and sorts it if its not).

I updated this, as properly tracking the sorted status of a COO was causing a regression, where the conversion was being done serially. Now that both paths are parallel, properly tracking the COO to be sorted does not result in a slowdown.